src/common/imagpng.cpp

```diff
@@ -737,6 +775,34 @@ bool wxPNGHandler::SaveFile( wxImage *image, wxOutputStream& stream, bool verbos
         png_set_pHYs( png_ptr, info_ptr, resX, resY, PNG_RESOLUTION_METER );
 
     png_set_sBIT( png_ptr, info_ptr, &sig_bit );
+
+    // save "Description" text chunk
+#if defined(PNG_WRITE_iTXt_SUPPORTED) || defined(PNG_WRITE_tEXt_SUPPORTED)
+    if (image->HasOption(wxIMAGE_OPTION_PNG_DESCRIPTION))
+    {
+        const wxString& description = image->GetOption(wxIMAGE_OPTION_PNG_DESCRIPTION);
+
+        png_text text;
+        text.key = const_cast<char*>(wxIMAGE_OPTION_PNG_DESCRIPTION_KEY);
+        text.lang = nullptr;
+        text.lang_key = nullptr;
+
+#ifdef PNG_WRITE_iTXt_SUPPORTED
+        text.compression = PNG_ITXT_COMPRESSION_NONE;
+        text.text = const_cast<char*>(description.utf8_str().data());
```

Comment:
Sorry, rereading this code again I realize that this is a critical problem as the pointer to the buffer data becomes dangling when the temporary buffer returned by `utf8_str()` is destroyed. This must be changed to

```cpp
const auto& buf = description.utf8_str();
text.text = const_cast<char*>(buf.data());
text.itxt_length = buf.length();
``` 

(the last line is just a micro-optimization avoiding an extra call to `strlen()`, but we *must* keep the buffer alive for long enough)